### PR TITLE
Enhancement/cal 708 delete account

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

- Adds `Delete Account` option to the `/settings/profile` page allowing users to delete their account
- Follows up to delete the user's Stripe account if they are a PRO plan user

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How should this be tested?

- [x] Go to `/settings/profile` and click on `Delete Account` at the bottom of the page to delete the user account
- [x] If they're a pro member, check if their account was deleted from stripe (I haven't been able to test this yet, so I need help testing this bit)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pages/api/user/membership.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (!req.body.teamId) {
+    console.log(req.body);
```
Comment (baileypumfleet) [inline pages/api/user/membership.ts:36]:
```suggestion

```
Should this be here?

**Review by pumfleet — COMMENTED** [review]
Code looks good, will test tomorrow

File: pages/api/user/[id].ts [hunk 2]
```diff
@@ -33,7 +33,12 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
   }
 
   if (req.method === "DELETE") {
-    return res.status(405).json({ message: "Method Not Allowed" });
+    await prisma.user.delete({
+      where: {
+        id: authenticatedUser?.id,
+      },
+    });
+    return res.status(200).json({ message: "User Deleted" });
```
Comment (emrysal) [inline pages/api/user/[id].ts:41]: I prefer (as you return no details about the entity just deleted, so you'd never use this response message):
```javascript
res.status(204).end()
```

File: pages/settings/profile.tsx [hunk 3]
```diff
@@ -112,6 +117,26 @@ function SettingsView(props: ComponentProps<typeof Settings> & { localeProp: str
     },
   });
 
+  const deleteAccount = async () => {
+    // Check if the user is pro ?? and then go ahead with stripe code, else ignore??
+    if (props.user.plan === "PRO") {
```
Comment (emrysal) [inline pages/settings/profile.tsx:122]:
```javascript
  const delPromises = [
    fetch("/api/user/" + props.user.id, {
      method: "DELETE",
      headers: {
        "Content-Type": "application/json",
      },
    });
  ];
  if (props.user.plan === "PRO") {
    delPromises.push(fetch("/api/user/stripe", {
      method: "DELETE",
    }));
  }
  await Promise.all(delPromises); // catch? What if something goes wrong?
  
  signOut({ callbackUrl: "/auth/logout" });
```

**Comment by alishaz-polymath** [comment]
> Code looks good, will test tomorrow

Yeah, I specifically need help with testing the Stripe flow as I was unable to test it out. I think it should work, but still need to confirm.

File: public/static/locales/en/common.json [hunk 4]
```diff
@@ -550,6 +550,9 @@
   "delete_event_type_description": "Are you sure you want to delete this event type? Anyone who you've shared this link with will no longer be able to book using it.",
   "delete_event_type": "Delete Event Type",
   "confirm_delete_event_type": "Yes, delete event type",
+  "delete_account": "Delete account",
+  "confirm_delete_account": "Yes, delete account",
+  "delete_account_confirmation_message":"Are you sure you want to delete your Cal.com account? Anyone who you&apos;ve shared your account link with will no longer be able to book using it and any preferences you have saved will be lost.",
```
Comment (baileypumfleet) [inline public/static/locales/en/common.json:555]:
```suggestion
  "delete_account_confirmation_message":"Are you sure you want to delete your Cal.com account? Anyone who you've shared your account link with will no longer be able to book using it and any preferences you have saved will be lost.",
```
This directly renders `&apos;` in the modal itself, so I believe we are safe to use an apostrophe directly in this string.

**Comment by alishaz-polymath** [comment]
**Forgot password test is failing, unrelated to this PR

**Review by zomars — APPROVED** [review]

**Review by emrysal — APPROVED** [review]

**Pull request merged**